gradle/lib/dependabot/gradle/file_updater.rb

```diff
@@ -112,26 +112,30 @@ def update_files_for_dep_set_change(buildfiles, old_req, new_req)
 
       def update_version_in_buildfile(dependency, buildfile, previous_req,
                                       requirement)
+        original_content = buildfile.content.dup
+
         updated_content =
-          buildfile.content.gsub(
-            original_buildfile_declaration(dependency, previous_req),
-            updated_buildfile_declaration(
-              dependency,
-              previous_req,
-              requirement
+          original_buildfile_declarations(dependency, previous_req).reduce(original_content) do |content, declaration|
+            content.gsub(
+              declaration,
+              updated_buildfile_declaration(
+                declaration,
+                previous_req,
+                requirement
+              )
             )
-          )
+          end
 
         raise "Expected content to change!" if updated_content == buildfile.content
 
         updated_file(file: buildfile, content: updated_content)
       end
 
-      def original_buildfile_declaration(dependency, requirement)
+      def original_buildfile_declarations(dependency, requirement)
         # This implementation is limited to declarations that appear on a
         # single line.
```

Comment:
A reminder that we'd benefit greatly from:
* https://github.com/dependabot/dependabot-core/issues/1164